src/lightning/fabric/utilities/distributed.py

```diff
@@ -389,3 +390,16 @@ def __init__(self, sampler: Union[Sampler, Iterable], *args: Any, **kwargs: Any)
     def __iter__(self) -> Iterator:
         self.dataset.reset()
         return (self.dataset[index] for index in super().__iter__())
+
+
+def _suggested_max_num_threads(num_processes: int = 1) -> int:
+    if num_processes < 1:
+        raise ValueError(f"`num_processes` should be >= 1, got {num_processes}.")
+    return max(1, _num_cpus_available() // num_processes)
+
+
+def _set_num_threads_if_needed(num_processes: int = 1) -> None:
+    if "OMP_NUM_THREADS" not in os.environ:
+        num_threads = _suggested_max_num_threads(num_processes)
+        torch.set_num_threads(num_threads)
+        os.environ["OMP_NUM_THREADS"] = str(num_threads)
```

Comment:
https://pytorch.org/docs/stable/notes/cpu_threading_torchscript_inference.html says that `set_num_threads` will ignore `OMP_NUM_THREADS` so maybe this line is not required.

That docs page talks about TorchScript but I think a lot of the information still applies.

Reply:
It's not that it ignores it, but rather `set_num_threads` has precedence over the env var. But I'm setting it more for completeness rather than necessity (the user may look for this env var?). But I could also leave it out, I have no preference over which way here.

Reply:
It's not that it ignores it, but rather `set_num_threads` has precedence over the env var. But I'm setting it more for completeness rather than necessity, so that the launched processes immediately get this setting. I could also leave it out, I don't mind.